File: src/main/java/org/cubexmc/metro/train/ConsistController.java
package org.cubexmc.metro.train;

import org.cubexmc.metro.Metro;

/**
 * Periodically enforces spacing and velocity alignment for a consist.
 */
public class ConsistController implements Runnable {

    private final Metro plugin;
    private final TrainConsist consist;

    public ConsistController(Metro plugin, TrainConsist consist) {
        this.plugin = plugin;
        this.consist = consist;
    }

    @Override
    public void run() {
        // Placeholder: keep-alive tick; spacing logic to be added later
    }
}

File: src/main/java/org/cubexmc/metro/train/ConsistController.kt
package org.cubexmc.metro.train

import org.cubexmc.metro.Metro

/**
 * Periodically enforces spacing and velocity alignment for a consist.
 */
class ConsistController(
    @Suppress("unused") private val plugin: Metro,
    @Suppress("unused") private val consist: TrainConsist,
) : Runnable {

    override fun run() {
        // Placeholder: keep-alive tick; spacing logic to be added later
    }
}
